client/components/tinymce/plugins/wpcom-view/plugin.js

```diff
@@ -854,23 +856,32 @@ function wpview( editor ) {
 
 	editor.addButton( 'wp_view_remove', {
 		tooltip: i18n.translate( 'Remove' ),
-		icon: 'dashicon dashicons-no',
+		icon: 'dashicon dashicons-trash',
 		onClick: function() {
 			selected && removeView( selected );
 		}
 	} );
 
 	editor.once( 'preinit', function() {
-		toolbar = editor.wp._createToolbar( [
+		editToolbar = editor.wp._createToolbar( [
 			'wp_view_edit',
 			'wp_view_remove'
 		] );
+		removeToolbar = editor.wp._createToolbar( [
+			'wp_view_remove'
+		] );
 	} );
 
 	editor.on( 'wptoolbar', function( event ) {
 		if ( selected ) {
+
+			// Checks for ids in the gallery shortcode
```

Comment:
This logic runs for any view which is selected (videos, embeds, contact forms), so it seems odd to include logic specific to galleries, particularly without first checking whether it is in fact a gallery view. 

Since we already have some [logic for `isEditable`](https://github.com/Automattic/wp-calypso/blob/ac6144fc04b57e40f22edbcca73f358a361870b8/client/components/tinymce/plugins/wpcom-view/views.js#L97-L99) in the `views.js` file, I think we could extend it for our use-case here. Perhaps instead of merely checking the presence of an `edit` function, we allow a view implementation to define an `isEditable` static function which is passed the content to test.

So `isEditable` in `views.js` becomes:

```js
isEditable( type, editor, content ) {
	if ( ! views[ type ] ) {
		return false;
	}

	if ( 'function' === typeof views[ type ].isEditable ) {
		return views[ type ].isEditable( editor, content );
	}

	return 'function' === typeof views[ type ].edit;
}
```

And then we'd need to pass `editor` and `content` in the implementation of the `wp_view_edit` button later in this file. We technically don't need `editor`, only `content`, but it'd be nice to pass consistent arguments between `edit` and `isEditable`.

The nice thing about this is that the `wp_view_edit` button implementation is already set up to hide itself if the view is not editable, so we don't need to create two separate toolbar instances.